robot_3d/src/joint_states_subscriber_3d.cpp

```diff
@@ -0,0 +1,101 @@
+#include <kdl/chain.hpp>
+#include <kdl/chainfksolver.hpp>
+#include <kdl/chainfksolverpos_recursive.hpp>
+#include <kdl/frames_io.hpp>
+
+#include <kdl_parser/kdl_parser.hpp>
+
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include "std_msgs/String.h"
+
+#include <tf_conversions/tf_kdl.h>
+#include <tf/transform_broadcaster.h>
+
+class JointMeasurementData
+{
+public:
+  JointMeasurementData(ros::NodeHandle* n)
```

Comment:
Since you are defining other functions outside the class, you can also define the ctor outside.